Fix golangci-lint issues

If you run `golangci-lint run`  locally you'll see the following warnings

```
pkg/cel/cel.go:34:3: SA1019: cel.Declarations is deprecated: use FunctionDecls and VariableDecls or FromConfig instead. (staticcheck)
		cel.Declarations(
		^
cmd/sink/filters/filters.go:214:23: SA1019: toolsWatch.NewRetryWatcher is deprecated: use NewRetryWatcherWithContext instead. (staticcheck)
	retryWatcher, err := toolsWatch.NewRetryWatcher("1", &cache.ListWatch{WatchFunc: watcher})
	                    ^
cmd/sink/filters/filters_test.go:148:25: SA1019: toolsWatch.NewRetryWatcher is deprecated: use NewRetryWatcherWithContext instead. (staticcheck)
			retryWatcher, err := toolsWatch.NewRetryWatcher("1", &cache.ListWatch{WatchFunc: watcher})
			                    ^
cmd/operator/internal/controller/kubearchiveconfig_controller.go:478:121: (*KubeArchiveConfigReconciler).reconcileA13e - result 0 (*knative.dev/eventing/pkg/apis/sources/v1.ApiServerSource) is never used (unparam)
func (r *KubeArchiveConfigReconciler) reconcileA13e(ctx context.Context, resources []sourcesv1.APIVersionKindSelector) (*sourcesv1.ApiServerSource, error) {
```

These warnings seem relatively straightforward to fix and we should especially move off of deprecated function calls

--- a/.github/workflows/golangci_lint.yml
+++ b/.github/workflows/golangci_lint.yml
@@ -4,6 +4,9 @@
 name: golangci-lint
 run-name: Linting Tests (golangci-lint)
 on:
+  schedule:
+    # 9:25 UTC every 1st of month
+    - cron: '25 09 1 * *'
   push:
     branches:
       - 'main'
@@ -19,7 +22,7 @@ jobs:
     uses: ./.github/workflows/log_run_details.yml
   golangci:
     # See https://github.com/golangci/golangci-lint-action
-    name: "golangci-lint security testing"
+    name: GolangCI Lint
     runs-on: ubuntu-latest
     steps:
       - name: Check out repository code
@@ -36,4 +39,22 @@ jobs:
         uses: golangci/golangci-lint-action@v6
         with:
           version: latest
-          only-new-issues: true
+          # When is scheduled, show all issues
+          only-new-issues: ${{ github.event_name != 'schedule' }}
+      - name: Notify golangci-lint failed
+        uses: jayqi/failed-build-issue-action@v1
+        if: failure() && github.event_name == 'schedule' && github.repository == 'kubearchive/kubearchive'
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          title-template: GolangCI Linting found errors
+          body-template: |
+            The scheduled workflow of GolangCI Lint found some problems. We run this scheduled
+            against the whole codebase, so when a PR is submitted authors do not need to deal with
+            problems they did not introduce. However we need to solve those problems from time to time.
+            The URL of the failed job is:
+
+            https://github.com/kubearchive/kubearchive/actions/runs/{{ runId }}
+
+            Run `golanci-lint` to see the errors locally, visit the worfklow run, make sure the problems
+            you found locally match. Investigate if they do not match. Solve the problems and open a
+            pull request to solve them.

--- a/cmd/operator/internal/controller/kubearchiveconfig_controller.go
+++ b/cmd/operator/internal/controller/kubearchiveconfig_controller.go
@@ -135,7 +135,7 @@ func (r *KubeArchiveConfigReconciler) Reconcile(ctx context.Context, req ctrl.Re
 	}
 
 	if len(resources) > 0 {
-		_, err = r.reconcileA13e(ctx, resources)
+		err = r.reconcileA13e(ctx, resources)
 		if err != nil {
 			return ctrl.Result{}, err
 		}
@@ -184,7 +184,7 @@ func (r *KubeArchiveConfigReconciler) cleanupK9eResources(ctx context.Context, k
 			return err
 		}
 
-		_, err = r.reconcileA13e(ctx, resources)
+		err = r.reconcileA13e(ctx, resources)
 		if err != nil {
 			return err
 		}
@@ -475,7 +475,7 @@ func (r *KubeArchiveConfigReconciler) desiredRoleBinding(kaconfig *kubearchivev1
 	return binding, nil
 }
 
-func (r *KubeArchiveConfigReconciler) reconcileA13e(ctx context.Context, resources []sourcesv1.APIVersionKindSelector) (*sourcesv1.ApiServerSource, error) {
+func (r *KubeArchiveConfigReconciler) reconcileA13e(ctx context.Context, resources []sourcesv1.APIVersionKindSelector) error {
 	log := log.FromContext(ctx)
 
 	log.Info("in reconcileApiServerSource")
@@ -488,20 +488,20 @@ func (r *KubeArchiveConfigReconciler) reconcileA13e(ctx context.Context, resourc
 		err = r.Update(ctx, source)
 		if err != nil {
 			log.Error(err, "Failed to update ApiServerSource")
-			return source, err
+			return err
 		}
 	} else if errors.IsNotFound(err) {
 		err = r.Create(ctx, source)
 		if err != nil {
 			log.Error(err, "Failed to create ApiServerSource")
-			return source, err
+			return err
 		}
 	} else {
 		log.Error(err, "Failed to reconcile ApiServerSource")
-		return source, err
+		return err
 	}
 
-	return source, nil
+	return nil
 }
 
 func (r *KubeArchiveConfigReconciler) deleteA13e(ctx context.Context) {

--- a/cmd/sink/filters/filters.go
+++ b/cmd/sink/filters/filters.go
@@ -211,7 +211,7 @@ func (f *Filters) Update() (UpdateStopper, error) {
 			metav1.SingleObject(metav1.ObjectMeta{Name: filtersCmName, Namespace: globalKey}),
 		)
 	}
-	retryWatcher, err := toolsWatch.NewRetryWatcher("1", &cache.ListWatch{WatchFunc: watcher})
+	retryWatcher, err := toolsWatch.NewRetryWatcherWithContext(context.Background(), "1", &cache.ListWatch{WatchFunc: watcher})
 	if err != nil {
 		return noopUpdateStopper, fmt.Errorf("Could not create a watcher for the %s ConfigMap: %s", filtersCmName, err)
 	}

--- a/cmd/sink/filters/filters_test.go
+++ b/cmd/sink/filters/filters_test.go
@@ -145,7 +145,7 @@ func TestHandleUpdates(t *testing.T) {
 			ch := make(chan watch.Event)
 			t.Cleanup(func() { close(ch) })
 			watcher := newMockWatcher(t, ch)
-			retryWatcher, err := toolsWatch.NewRetryWatcher("1", &cache.ListWatch{WatchFunc: watcher})
+			retryWatcher, err := toolsWatch.NewRetryWatcherWithContext(t.Context(), "1", &cache.ListWatch{WatchFunc: watcher})
 			if err != nil {
 				assert.FailNow(t, "Could not create a watcher")
 			}

--- a/pkg/cel/cel.go
+++ b/pkg/cel/cel.go
@@ -9,7 +9,8 @@ import (
 	"strings"
 
 	"github.com/google/cel-go/cel"
-	"github.com/google/cel-go/checker/decls"
+	"github.com/google/cel-go/common/decls"
+	"github.com/google/cel-go/common/types"
 	"github.com/google/cel-go/common/types/ref"
 	celext "github.com/google/cel-go/ext"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
@@ -24,17 +25,17 @@ func CompileOrCELExpression(exprs ...string) (*cel.Program, error) {
 }
 
 func CompileCELExpr(expr string) (*cel.Program, error) {
-	mapStrDyn := decls.NewMapType(decls.String, decls.Dyn)
+	mapStrDyn := types.NewMapType(types.StringType, types.DynType)
 	env, err := cel.NewEnv(
 		celext.Strings(),
 		celext.Encoders(),
 		celext.Sets(),
 		celext.Lists(),
 		celext.Math(),
-		cel.Declarations(
-			decls.NewVar("metadata", mapStrDyn),
-			decls.NewVar("spec", mapStrDyn),
-			decls.NewVar("status", mapStrDyn),
+		cel.VariableDecls(
+			decls.NewVariable("metadata", mapStrDyn),
+			decls.NewVariable("spec", mapStrDyn),
+			decls.NewVariable("status", mapStrDyn),
 		),
 	)
 	if err != nil {

